From f756dca6b96ec4c208316ce595209c53e45eb427 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Thu, 6 Jul 2017 06:41:49 -0400
Subject: [PATCH] [1/2] Enable heap overflow check for stdin for mpc files

Enabling seekable streams is required to ensure checking the blob size
works when an image is streamed on stdin.

origin: https://github.com/ImageMagick/ImageMagick/commit/b007dd3a048097d8f58949297f5b434612e1e1a3
bug: https://github.com/ImageMagick/ImageMagick/issues/556
bug-debian: https://bugs.debian.org/867896

(cherry picked from commit b007dd3a048097d8f58949297f5b434612e1e1a3)
---
 coders/mpc.c | 2 ++
 1 file changed, 2 insertions(+)

Index: imagemagick-6.7.7.10/coders/mpc.c
===================================================================
--- imagemagick-6.7.7.10.orig/coders/mpc.c	2017-07-21 10:01:15.197294288 -0400
+++ imagemagick-6.7.7.10/coders/mpc.c	2017-07-21 10:01:15.193294288 -0400
@@ -990,6 +990,7 @@ ModuleExport size_t RegisterMPCImage(voi
   entry=SetMagickInfo("CACHE");
   entry->description=ConstantString("Magick Persistent Cache image format");
   entry->module=ConstantString("CACHE");
+  entry->seekable_stream=MagickTrue;
   entry->stealth=MagickTrue;
   (void) RegisterMagickInfo(entry);
   entry=SetMagickInfo("MPC");
@@ -997,6 +998,7 @@ ModuleExport size_t RegisterMPCImage(voi
   entry->encoder=(EncodeImageHandler *) WriteMPCImage;
   entry->magick=(IsImageFormatHandler *) IsMPC;
   entry->description=ConstantString("Magick Persistent Cache image format");
+  entry->seekable_stream=MagickTrue;
   entry->module=ConstantString("MPC");
   (void) RegisterMagickInfo(entry);
   return(MagickImageCoderSignature);
